stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpConstants.java

```diff
@@ -305,4 +307,16 @@
     public static final String CLIENT_EP_RETRY = "retry";
     public static final String RETRY_COUNT = "count";
     public static final String RETRY_INTERVAL = "interval";
+
+    public static final int SERVICE_ENDPOINT_PROTOCOL_INDEX = 0;
+
+    //Remote struct indexes
+    public static final int REMOTE_STRUCT_INDEX = 2;
+    public static final int REMOTE_HOST_INDEX = 0;
+    public static final int REMOTE_PORT_INDEX = 0;
+
+    //Local struct indexes
+    public static final int LOCAL_STRUCT_INDEX = 3;
+    public static final int LOCAL_HOST_INDEX = 0;
+    public static final int LOCAL_PORT_INDEX = 0;
```

Comment:
Do we need to be using these now? For structs, can't we just deal with `Struct`s and then convert those into `BStruct`s?

Reply:
If we do this BStruct to struct conversion for each request, it may introduce performance bottleneck.